wlauto/result_processors/uxperf.py

```diff
@@ -0,0 +1,229 @@
+from wlauto.utils.fps import FpsProcessor
+from wlauto.utils.types import numeric
+
+try:
+    import pandas as pd
+except ImportError:
+    pd = None
+
+
+class UxPerfResultProcessor(ResultProcessor):
+
+    name = 'uxperf'
+    description = '''
+    Parse logcat for UX_PERF markers to produce performance metrics for
+    workload actions using specified instrumentation.
+
+    An action represents a series of UI interactions to capture.
+
+    NOTE: The UX_PERF markers are turned off by default and must be enabled in
+    a agenda file by setting dumpsys_enabled for the workload to true.
+    '''
+
+    parameters = [
+        Parameter('add_timings', kind=bool, default=True,
+                  description='''
+                  Add per-action timings to result metrics.'
+                  '''),
+        Parameter('add_frames', kind=bool, default=False,
+                  description='''
+                  Add per-action frame statistics to result metrics.
+                  i.e. fps, frame_count, jank and not_at_vsync.
+
+                  NOTE: This option requires the fps instrument to be enabled.
+                  '''),
+        Parameter('drop_threshold', kind=numeric, default=5,
+                  description='''
+                  Data points below this FPS will be dropped as they do not
+                  constitute "real" gameplay. The assumption being that while
+                  actually running, the FPS in the game will not drop below X
+                  frames per second, except on loading screens, menus, etc,
+                  which should not contribute to FPS calculation.
+                  '''),
+    ]
+
+    def initialize(self, context):
+        if not pd or LooseVersion(pd.__version__) < LooseVersion('0.13.1'):
+            message = ('uxperf result processor requires pandas Python package '
+                       '(version 0.13.1 or higher) to be installed.\n'
+                       'You can install it with pip, e.g. "sudo pip install pandas"')
+            raise ResultProcessorError(message)
+
+    def export_iteration_result(self, result, context):
+        parser = UxPerfParser(context)
+
+        logfile = os.path.join(context.output_directory, 'logcat.log')
+        framelog = os.path.join(context.output_directory, 'frames.csv')
+
+        parser.parse(logfile)
+
+        if self.add_timings:
+            parser.add_action_timings()
+
+        if self.add_frames:
+            parser.parse_frames(framelog, self.drop_threshold)
+
+
+class UxPerfParser(object):
+    '''
+    Parses logcat messages for UX Performance markers.
+
+    UX Performance markers are output from logcat under a debug priority. The
+    logcat tag for the marker messages is UX_PERF. The messages associated with
+    this tag consist of a name for the action to be recorded and a timestamp.
+    These fields are delimited by underscores. e.g.
+
+    <TAG>   : <MESSAGE>
+    UX_PERF : gestures_swipe_left_start_861975087367
```

Comment:
Would it not be easier/clearer to have the timestamp separated from the tag by a space?